eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RoutingExtension.php

```diff
@@ -78,14 +88,30 @@ public function getPath(object $name, array $parameters = [], bool $relative = f
     {
         $referenceType = $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH;
 
-        return $this->generateUrlForObject($name, $parameters, $referenceType);
+        return $this->tryGeneratingUrlForObject($name, $parameters, $referenceType);
     }
 
     public function getUrl(object $name, array $parameters = [], bool $schemeRelative = false): string
     {
         $referenceType = $schemeRelative ? UrlGeneratorInterface::NETWORK_PATH : UrlGeneratorInterface::ABSOLUTE_URL;
 
-        return $this->generateUrlForObject($name, $parameters, $referenceType);
+        return $this->tryGeneratingUrlForObject($name, $parameters, $referenceType);
+    }
+
+    private function tryGeneratingUrlForObject(object $object, array $parameters, int $referenceType): string
+    {
+        try {
+            return $this->generateUrlForObject($object, $parameters, $referenceType);
+        } catch (NotFoundException $e) {
+            return '';
+        } catch (Throwable $e) {
+            $this->logger->warning(
+                'Url could not be generated.',
+                ['exception' => $e]
+            );
+
+            return '';
```

Comment:
Is it acceptable for us to **always** discard the error? Invalid route usually throws an error in Symfony (and it used to do so for us). This is technically a change in behavior.

Can we look for a different, alternative approach? A parameter, a state?

We could create a Twig token to start and stop a "relaxed" rendering?

```twig
{% relaxed_start %}
    {{ ez_route(...) }}
{% relaxed_end %}
```

Reply:
> Is it acceptable for us to **always** discard the error? Invalid route usually throws an error in Symfony (and it used to do so for us). This is technically a change in behavior.
> 
> Can we look for a different, alternative approach? A parameter, a state?
> 
> We could create a Twig token to start and stop a "relaxed" rendering?
> 
> ```
> {% relaxed_start %}
>     {{ ez_route(...) }}
> {% relaxed_end %}
> ```

Hey @Steveb-p this was my idea. If this should be handled in Twig in a better way, we're of course opened to suggestions. I look at this from API perspective and from API perspective trying to render URL in case of API NotFound Exception doesn't make sense. The URL for main language will be incorrect in that context.

Reply:
Concerning the above, don't you think that `Url could not be generated` is not informative enough? Wouldn't it be better to try logging the actual exception messages coming directly from `UrlGeneratorInterface::generate` method?

Reply:
> Adding relaxed rendering would mean we have to change all the templates utilizing `ez_route` now, right?

Yes, that would be the effect.

> I look at this from API perspective and from API perspective trying to render URL in case of API NotFound Exception doesn't make sense.

I understand why we need it, I am just not happy about how broad the effect of this change will be. We are basically making `ez_path` and family functions somewhat unsafe, different from what they used to be.

I could agree to having this function behave differently if a Content object is not found, or a Content that the Location points to is not found. That makes sense to me. However, swallowing any error is too much. Normal route references should still throw a hard error.

I think our solution should be to only handle the special case in question, and allow any other errors and exceptions to propagate normally. They are potentially a logic error and should not be hidden away.

Reply:
> I could agree to having this function behave differently if a Content object is not found, or a Content that the Location points to is not found. That makes sense to me. However, swallowing any error is too much. Normal route references should still throw a hard error.
> 
> I think our solution should be to only handle the special case in question, and allow any other errors and exceptions to propagate normally. They are potentially a logic error and should not be hidden away.

@Steveb-p so, if we handle just `NotFound` and leave out logging + handling Throwable, would that be okay for now?

Reply:
I believe that would be the best approach. We can handle our own errors, but we shouldn't touch the ones generated from normal Symfony routes, in my opinion.

Reply:
> I believe that would be the best approach. We can handle our own errors, but we shouldn't touch the ones generated from normal Symfony routes, in my opinion.

@barw4 ok, let's cut the last catch and logger changes as you initially suggested yourself. You were right :grinning:

Reply:
@alongosz @Steveb-p resolved in  https://github.com/ezsystems/ezplatform-kernel/pull/389/commits/791b764e227720160ca4d6309a28c8a4b0018576